File: ios/Sources/CapacitorNfcAccessibilityCheckPlugin/CapacitorNfcAccessibilityCheckPlugin.swift
import Foundation
import Capacitor
import CoreNFC

@objc(CapacitorNfcAccessibilityCheckPlugin)
public class CapacitorNfcAccessibilityCheckPlugin: CAPPlugin, CAPBridgedPlugin {

    public let identifier = "CapacitorNfcAccessibilityCheckPlugin"
    public let jsName = "CapacitorNfcAccessibilityCheck"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "isSupported", returnType: CAPPluginReturnPromise)
    ]

    private let implementation = CapacitorNfcAccessibilityCheck()

    @objc func isSupported(_ call: CAPPluginCall) {
        let supported = implementation.isSupported()
        call.resolve([
            "result": supported // JS receives boolean in "value"
        ])
    }
}

File: android/src/main/java/com/capacitor/nfc/accessibility/check/CapacitorNfcAccessibilityCheckPlugin.java
package com.capacitor.nfc.accessibility.check;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "CapacitorNfcAccessibilityCheck")
public class CapacitorNfcAccessibilityCheckPlugin extends Plugin {

    private CapacitorNfcAccessibilityCheck implementation;

    @Override
    public void load() {
        implementation = new CapacitorNfcAccessibilityCheck(getContext());
    }

    @PluginMethod
    public void isSupported(PluginCall call) {
        boolean supported = implementation.isSupported();

        JSObject ret = new JSObject();
        ret.put("result", supported);
        call.resolve(ret);
    }
}
